core/src/helpers/extractEventFromTransaction.ts

```diff
@@ -0,0 +1,11 @@
+import getProvider from '../provider';
+import { ethers } from 'ethers';
+
+const extractEventFromTransaction = async (network: string, transactionHash: string, event: string) => {
+    const provider = await getProvider(network);
+    const receipt = await provider.getTransactionReceipt(transactionHash);
```

Comment:
This will fetch logs unparsed, instead, you can wait for the transaction to get the events parsed according to ABI via [`transaction.wait`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt). Or, which will be more cumbersome: `tx.logs.map((log) => contract.interface.parseLog(log))`.

(parsing will not only help with automatically parsed event names, but also with automatically parsed values, so you don't have to use `ethers.utils.defaultAbiCoder.decode` in the parent)

Therefore I suggest to pass transaction here instead of the hash, wait for it and return list of events

Reply:
Well, you're right that my suggestion wouldn't improve some particular edge-case, but I believe it would still return you an object with all events, even the ones that wasn't parsed. Also, if you wish to improve this particular edge case as well, you can add optional `contractInterfaces` argument to supply other abis and parse it manually via `tx.logs.map((log) => contract.interface.parseLog(log))` provided above. I just suggest to make it nice and reusable, not just hack/hardcode it every time you need to parse a transaction event.